### ISSUE TITLE
SPFx webpart is throwing error: The user or administrator has not consented to use the application with ID

### ISSUE THREAD

**Issue by markgort86** [SharePoint/sp-dev-docs#8304]
### What type of issue is this?

other

### What SharePoint development model, framework, SDK or API is this about?

SharePoint Add-ins

### Target SharePoint environment

SharePoint Online

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- SPFx version: 1.11.0
- Node.js version: 8.11.3

### Issue description

**Expected or Desired Behavior**
SPFx webpart is working without the error: The user or administrator has not consented to use the application with ID.

**Observed Behavior**
This webpart is fully functional on our test tenant, so we believe the code in this SharePoint SPFx webpart should not be the problem. We cannot find any difference in the configuration that might be the cause. In this SPFx webpart we have configured the following webApiPermissions:

```JSON
    "webApiPermissionRequests": [
      {
        "resource": "Windows Azure Active Directory",
        "scope": "User.Read"
      },
      {
        "resource": "Office 365 SharePoint Online",
        "scope": "User.ReadWrite.All"
      },
      {
        "resource": "Office 365 SharePoint Online",
        "scope": "AllSites.FullControl"
      },
      {
        "resource": "Clusters",
        "scope": "user_impersonation"
      }
    ],
````

In the app registration settings, the following permissions are configured for the SharePoint Online Client Extensibility Web Application Principal.

![API permissions](https://user-images.githubusercontent.com/52573368/179201898-70d72557-dad6-43d8-bc36-c1d61ab41b3c.PNG)
 
**Received console message below when running this SPFx solution**
InteractionRequiredAuthError: AADSTS65001: The user or administrator has not consented to use the application with ID '815bf0d2-1cc5-4be2-8ab1-740271020568' named 'SharePoint Online Client Extensibility Web Application Principal'. Send an interactive authorization request for this user and resource.

If there are more details needed to give us some feedback, please let us know it by placing a reaction to this GitLab post.

**Comment by AJIXuMuK** [comment]
@markgort86 - could you please check the ID of the Azure App is the same as listed in the error?
If yes - please, delete the app and revisit the API Access page.
It should recreate the app and re-assign the permissions.

Thanks!

**Comment by markgort86** [comment]
The ID of the app is corresponding with the ID of the following App registration: SharePoint Online Client Extensibility Web Application Principal. 

@AJIXuMuK Is it a good idea to delete this one if there are more apps using it (in a production environment)? We cannot have any downtime for any of them, at during business hours. 

What I did so far using a fresh and clean Microsoft 365 tenant:

- I uploaded an SPFx webpart to the app catalog (that uses the Microsoft Graph). Both registrations are created (principal and helper) after deployment and enabling of this webpart. I granted one of the API permissions for Microsoft Graph, then deleted the app registrations and afterwards also the app from the app catalog. I then perfomed these steps again. Both registrations were recreated once again. 
- I tested an another scenario with an installed app with granted permissions. I deleted only the app registration with the following name: SharePoint Online Client Extensibility Web Application Principal. After a while the app registration was back in the Azure AD app registrations. But the granted permissions were lost.
